domain/src/main/java/org/oppia/android/domain/feedbackreporting/FeedbackReportManagementController.kt

```diff
@@ -0,0 +1,346 @@
+      reportSubmissionUtcOffset = report.reportSubmissionUtcOffsetHours,
+      userSuppliedFeedback = createGaeUserSuppliedFeedback(report.userSuppliedInfo),
+      systemContext = getSystemContext(report.systemContext),
+      deviceContext = getDeviceContext(report.deviceContext),
+      appContext = getAppContext(report.appContext)
+    )
+    // TODO(#76): Implement a callback for success / failure handling once a network retry policy is
+    // established.
+    feedbackReportingService.postFeedbackReport(gaeFeedbackReport).execute()
+  }
+
+  /** Removes the first cached report from the store. */
+  fun removeFirstCachedReport() {
+    feedbackReportDataStore.storeDataAsync(updateInMemoryCache = true) { feedbackReportDatabase ->
+      return@storeDataAsync feedbackReportDatabase.toBuilder().removeReports(0).build()
+    }.invokeOnCompletion {
+      it?.let {
+        consoleLogger.e(
+          FEEDBACK_REPORT_MANAGEMENT_CONTROLLER_TAG,
+          "Failed to remove report from store",
+          it
+        )
+      }
+    }
+  }
+
+  // Helper function to store a single report in memory on the device.
+  private fun storeFeedbackReport(report: FeedbackReport) {
+    feedbackReportDataStore.storeDataAsync(updateInMemoryCache = true) { feedbackReportDatabase ->
+      return@storeDataAsync feedbackReportDatabase.toBuilder().addReports(report).build()
+    }.invokeOnCompletion {
+      it?.let {
+        consoleLogger.e(
+          FEEDBACK_REPORT_MANAGEMENT_CONTROLLER_TAG,
+          "Failed to store feedback report",
+          it
+        )
+      }
+    }
+  }
+
+  // Creates the Moshi data class that gets sent in the service for the information collected from
+  // user responses.
+  private fun createGaeUserSuppliedFeedback(
+    userSuppliedFeedback: UserSuppliedFeedback
+  ): GaeUserSuppliedFeedback {
+    val reportType = userSuppliedFeedback.reportTypeCase
+    when (reportType) {
+      ReportTypeCase.SUGGESTION -> {
+        val suggestion = userSuppliedFeedback.suggestion
+        return GaeUserSuppliedFeedback(
+          reportType = reportType.name.toLowerCase(),
+          category = suggestion.suggestionCategory.name.toLowerCase(),
+          userFeedbackSelectedItems = null,
+          userFeedbackOtherTextInput = suggestion.userSubmittedSuggestion
+        )
+      }
+      ReportTypeCase.ISSUE -> {
+        return createGaeUserSuppliedFeedbackForIssue(reportType.name, userSuppliedFeedback.issue)
+      }
+      ReportTypeCase.CRASH -> {
+        val crash = userSuppliedFeedback.crash
+        return GaeUserSuppliedFeedback(
+          reportType = reportType.name.toLowerCase(),
+          category = crash.crashLocation.name.toLowerCase(),
+          userFeedbackSelectedItems = null,
+          userFeedbackOtherTextInput = crash.crashExplanation
+        )
+      }
+      else -> throw IllegalArgumentException(
+        "Encountered unexpected feedback report type: ${userSuppliedFeedback.reportTypeCase.name}"
```

Comment:
```suggestion
        "Encountered unexpected feedback report type: ${userSuppliedFeedback.reportTypeCase}"
```

Optional: the enum-to-string conversion should keep this error message the way you intend it to be.

Ditto elsewhere.

Reply:
Done